### ISSUE TITLE
SSH port information is not inherited from cluster 

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#829]
Originally reported of Google Code with ID 773.

```
Unlike other NDparams, the SSH port information is not inherited from the cluster config, but the cluster holds the default value for the SSH ports. In order to avoid duplicate code first checking in the nodegroup's config and then looking for the cluster default, add a function to the Config writer to retrieve the SSH port given a node group UUID.

Also update the node_verify RPC to use this function.
```
Originally added on 2014-03-24 13:55:44 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Technically, the SSH port number is inherited from the cluster configuration, but we never allow it to be set. I'd suggest to make a method that returns a dictionary mapping group UUIDs to their SSH ports. This could be directly passed to the RPC calls, and it's easy to get the port for a single group. Under the hood, this function would call _UnlockedGetNdGroupParams (which would also need to be added, currently we have only the locked variant) for all groups to get the ports. This would save calls to the WConfd for every group, and would be safe in the case we'd ever decide to allow cluster SSH ports.

(Maybe it should go to 2.11, as this is a kind of shortcomming?)
```
Originally added on 2014-03-24 14:31:42 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Some more TODOs here:
- refactor NodeVerfiy (full and light) to use the new functions as well.
```
Originally added on 2014-03-25 09:44:18 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed in 2.16 with this:

commit acb218be9ee5b85b14a7b800d1c12df05f7ccadb
Author: Helga Velroyen <helgav@google.com>
Date:   Tue Oct 27 15:17:06 2015 +0100

    Use ssconf for SSH ports in NodeVerify
    
    This fixes issue 773. For a while already, SSH ports are
    available via ssconf. This enables us to simplify the
    RPC signature of NodeVerify.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

```
Originally added on 2015-10-27 16:03:29 +0000 UTC.

Mentioned Issue with GoogleCode ID 773 has been migrated to GitHub with ID #829.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2016-01-11 15:06:33 +0000 UTC.

Changed State: Released
